controller/client/client.go

```diff
@@ -66,77 +90,226 @@ func normalizeConfig(conf *Config) error {
 	return nil
 }
 
-type client struct {
-	url        string
-	authToken  string
-	httpClient *http.Client
+// ListRules returns the list of rules
+func (c *Client) ListRules(filter *api.RuleFilter) (*api.RulesSet, error) {
+	var rulesSet api.RulesSet
+
+	path := rulesPath
+	query := c.filterToStringQuery(filter)
+
+	if query != "" {
+		path += query
+	}
+
+	if err := c.doRequest("GET", path, nil, nil, &rulesSet, http.StatusOK); err != nil {
+		logrus.WithError(err).Warn("Failed to retrieve rules from controller")
+		return nil, err
+	}
+
+	return &rulesSet, nil
 }
 
-func (c *client) ListRules(filter *api.RuleFilter) (*api.RulesSet, error) {
-	var rulesSet api.RulesSet
+// CreateRules creates the rules
+func (c *Client) CreateRules(rulesSet *api.RulesSet) (interface{}, error) {
+	return c.setRules("POST", rulesSet, http.StatusCreated)
+}
 
-	u, err := url.Parse(c.url + "/v1/rules")
-	if err != nil {
-		return &rulesSet, err
+// UpdateRules updates the rules
+func (c *Client) UpdateRules(rulesSet *api.RulesSet) (interface{}, error) {
+	return c.setRules("PUT", rulesSet, http.StatusOK)
+}
+
+// DeleteRules deletes the rules
+func (c *Client) DeleteRules(filter *api.RuleFilter) ([]byte, error) {
+
+	path := rulesPath
+	query := c.filterToStringQuery(filter)
+
+	if query != "" {
+		path += query
 	}
 
-	query := u.Query()
-	for _, id := range filter.IDs {
-		query.Add("id", id)
+	if err := c.doRequest("DELETE", path, nil, nil, nil, http.StatusOK); err != nil {
+		logrus.WithError(err).Warn("Failed to delete rules in controller")
+		return nil, err
 	}
 
-	for _, tag := range filter.Tags {
-		query.Add("tag", tag)
+	return []byte("Request Completed"), nil
```

Comment:
Why are we returning `[]byte("Request Completed")` or `nil`? If "Request Completed" needs to be output somewhere, we should just check whether error is `nil` or not.
```
err := controllerClient.DeleteRules(filter)
if err == nil {
   fmt.Println("Request Completed")
}